[JENKINS-35458] Migrate from using bouncy castle to bouncycastle-api plugin

alvarolobato opened the pull request:
[JENKINS-35458](https://issues.jenkins-ci.org/browse/JENKINS-35458)

Dependency to multiple Bouncy Castle versions from jenkins core and plugins is causing problems due to the binary incompatibility between versions, the different supported algorithms, etc.

Plugins should be migrated to use a common API and BC version in order to avoid these problems, see [bouncycastle-api plugin](https://github.com/jenkinsci/bouncycastle-api/blob/master/README.md)
- Plugin parent had to be bumped in order to bee able to execute the new test using JenkinsRule.
- The minimum version of jenkins core supported by this plugin parent pom is 1.580.
  @reviewbybees

alvarolobato committed:
Migrate to bouncycastle-api plugin

amuniz commented:
Build failing because of `-Xdoclint:none` in Java 7. Either the job must be re-configured to use JDK 8 or put the `-Xdoclint:none` option set inside a profile with JDK version based activation.

alvarolobato committed:
fix javadoc lint for idk 7

alvarolobato commented:
@amuniz yeap, saw it in the car and couldn't change it. Uploaded change in a profile, I think is better to compile in JDK 7 as it is still supported.

alvarolobato commented:
OMG 28 FindBugs bugs after the changes, prior to this change FindBugs wasn't being run. I'll have a look tomorrow but I don't think we can afford fixing this right now and will have to defer it to another PR, disabling FB by now, as it was prior to this PR.

alvarolobato committed:
- Disabled Findbugs failures
- Bumped version to 1.609, it is the oldest version supported by bouncycastle-api plugin

alvarolobato commented:
Had to disable findbugs as it was before the pom parent change.

rsandell commented:
:bee:

recena commented:
🐝

recena commented:
@reviewbybees done

alvarolobato commented:
@martinsawicki @snallami could you spare a few minutes to have a look a this PR?
I'm proposing these changes in order to ensure that the plugin works correctly with all the versions of jenkins and different Bouncy Castle versions. 

Thanks

alvarolobato committed:
Bump to last version of BC API

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

recena commented:
@martinsawicki @snallami Any feedback?

martinsawicki commented:
thanks. Asking someone on our team to tak a look.
Note we're almost finished with a whole new version of this plugin, which would support Azure's new resource model, rather than the "Classic" model in this plugin.
So far available in the https://github.com/jenkinsci/azure-slave-plugin/tree/ARM-dev branch.

recena commented:
@martinsawicki Thanks for your feedback.

alvarolobato commented:
@martinsawicki @snallami did you or your team have a change to look at this? thanks.

arroyc commented:
@alvarolobato  .. sorry for belated response ...we will review it .. but would like you to try our new ARM-dev branch which comes with lot of these fixes already

clguiman commented:
@alvarolobato This plugin was deprecated. You should use Azure VM Agents plugin ( https://plugins.jenkins.io/azure-vm-agents). The new plugin doesn't depend on the bouncy castle so I won't move your PR to the new plugin repo
